graph/resolver/schema.resolvers.go

```diff
@@ -145,12 +154,17 @@ func (r *mutationResolver) RemoveFlair(ctx context.Context, id string) (*model.F
 		return nil, fmt.Errorf("Already removed")
 	}
 
-	// TODO: This only allows the requesting user to remove flair for
-	// themselves. But an admin or moderator could potentially remove flair on
-	// behalf of another user. To add this functionality remove the following
-	// line and add authorization/permission logic.
+	// An admin or moderator could potentially remove flair on
+	// behalf of another user.
 	if currentUser.UserID != flair.UserID {
-		return nil, fmt.Errorf("Unauthorized")
+		// Check if a moderator is trying to removeFlair
+		mod, err := r.DAOManager.GetModeratorByUserID(ctx, currentUser.UserID)
+		if err != nil {
+			return nil, fmt.Errorf("failed to fetch moderator")
```

Comment:
I think we should just return unauthorized in these cases -- we wouldn't want to expose this to the client I think